src/main/java/com/twcable/jackalope/impl/jcr/SessionImpl.java

```diff
@@ -265,24 +287,25 @@ public void exportDocumentView(String absPath, OutputStream out, boolean skipBin
 
     @Override
     public void setNamespacePrefix(String prefix, String uri) throws NamespaceException, RepositoryException {
+        namespaceRegistry.registerNamespace(prefix, uri);
     }
 
 
     @Override
     public String[] getNamespacePrefixes() throws RepositoryException {
-        return new String[0];
+        return namespaceRegistry.getPrefixes();
```

Comment:
Should probably add (particularly since the implementation is in a private static class) a couple of tests in SessionImplSpecification for these.